lib/ansible/modules/cloud/amazon/elasticache.py

```diff
@@ -481,31 +481,31 @@ def _get_nodes_to_remove(self):
 
 
 def main():
-    argument_spec = ec2_argument_spec()
+    """ elasticache ansible module """
+    argument_spec             = ec2_argument_spec()
     argument_spec.update(dict(
-        state                 ={'required': True, 'choices': ['present', 'absent', 'rebooted']},
-        name                  ={'required': True},
-        engine                ={'required': False, 'default': 'memcached'},
-        cache_engine_version  ={'required': False},
-        node_type             ={'required': False, 'default': 'cache.m1.small'},
-        num_nodes             ={'required': False, 'default': None, 'type': 'int'},
+        state                 = {'required': True, 'choices': ['present', 'absent', 'rebooted']},
+        name                  = {'required': True},
+        engine                = {'required': False, 'default': 'memcached'},
+        cache_engine_version  = {'required': False, 'default': ""},
+        node_type             = {'required': False, 'default': 'cache.t2.small'},
+        num_nodes             = {'required': False, 'default': 1, 'type': 'int'},
         # alias for compat with the original PR 1950
-        cache_parameter_group ={'required': False, 'default': None, 'aliases': ['parameter_group']},
-        cache_port            ={'required': False, 'type': 'int'},
-        cache_subnet_group    ={'required': False, 'default': None},
-        cache_security_groups ={'required': False, 'default': [], 'type': 'list'},
-        security_group_ids    ={'required': False, 'default': [], 'type': 'list'},
-        zone                  ={'required': False, 'default': None},
-        wait                  ={'required': False, 'type' : 'bool', 'default': True},
-        hard_modify           ={'required': False, 'type': 'bool', 'default': False}
-    )
-    )
+        cache_parameter_group = {'required': False, 'default': "", 'aliases': ['parameter_group']},
+        cache_port            = {'required': False, 'type': 'int', 'default': 11211},
+        cache_subnet_group    = {'required': False, 'default': ""},
+        cache_security_groups = {'required': False, 'default': [], 'type': 'list'},
+        security_group_ids    = {'required': False, 'default': [], 'type': 'list'},
+        zone                  = {'required': False, 'default': ""},
+        wait                  = {'required': False, 'default': True, 'type': 'bool'},
+        hard_modify           = {'required': False, 'default': False, 'type': 'bool'}
+    ))
 
     module = AnsibleModule(
         argument_spec=argument_spec,
     )
 
-    if not HAS_BOTO:
+    if not HAS_BOTO3:
         module.fail_json(msg='boto required for this module')
 
     region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
```

Comment:
This is being done where the connection is created too. Maybe better to get rid of this and keep it there, or vice versa? You could create the connection here in main and then pass it to the ElastiCacheManager as well. What makes most sense?